CONTRIBUTING.md

```diff
@@ -1,9 +1,203 @@
 
-<3
+
+## Enhancements vs bugs
+Enhancements are:
+* New features implementation.
+* Code refactoring.
+* Coding rules or coding style improvements.
+* Code comments improvement.
+* Documentation work.
+Bugs are:
+* Issues raised internally or externally by application users.
+* Internally (from the ARM mbed team) created issues from the Continuous Integration pipeline and build servers.
+* Issues detected using automation tools such as compilers, sanitizers, static code and analysis tools.
+
+## Workflow
+### Forking and cloning this repository
+First [fork](https://help.github.com/articles/fork-a-repo/) this repository in GitHub, then clone it locally with:
+```
+$ git clone <repo-link>
+```
+Now you can create separate branches in the forked repository and prepare pull requests with changes.
+### Issues solving
+Simple work-flow issue solving process may contain below steps:
+1. Issue filed (by any user).
+2. Proper label assigned by gate-keeper.
+3. Bug-fixer forked and cloned.
+4. Optional clarifications made using the Issues tab's Comment section.
+5. Pull request with fix created.
+6. Pull request reviewed by others.
+7. All code review comments handled.
+8. Pull request accepted by gate-keeper.
+9. Pull request merged successfully.
+
+### Code review
+The code review process is designed to catch both style and domain specific issues. It is also designed to follow and respect the _definition of done_. Please make sure your code follows the style of the source code you are modifying. Each pull request must be reviewed by the gate-keeper before we can merge it to the master branch.
+## Issues and bug reporting
+Please report all bugs using the Issues tab on the GitHub web-page. It will help us to collaborate on issues more promptly. One of our gate-keepers (developers responsible for quality and the repository) will review the issue and assign a label such as _bug_, _enhancement_, _help wanted_ or _wontfix_.
+# How to contribute
+You can either file a bug, help fix a bug or propose a new feature (or enhancement) and implement it yourself. If you want to contribute please:
+* Bug reports: File a bug report in the Issues tab of this repo to let us know there are problems with our code.
+  * Make sure your bug report contains a simple description of the problem.
+  * Include information about the host computer's configuration and OS or VM used.
+  * Include information about the application's version. All applications should have at least a ``--version`` switch you can use to check the version.
+  * Copy/paste useful console dumps and configuration files' content. Please use [fenced code blocks](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks) to encapsulate code snippets.
+* New features or bug fix: Create a pull request with your changes.
+* General feedback: Give your feedback by posting your comments on existing pull requests and issues.
+
+## Simple workflow for bugifx
```

Comment:
This looks like a duplication of the information in the workflow higher up - should rationalise into one